Description: fix compile error due to implicit function declaration
Index: uucp/unix/pipe.c
===================================================================
--- uucp.orig/unix/pipe.c	2024-03-16 19:03:45.075631292 +0100
+++ uucp/unix/pipe.c	2024-04-13 00:46:14.841462245 +0200
@@ -37,6 +37,10 @@
 
 #include <errno.h>
 
+/* needed for waitpid (XXX might be done with #if) */
+#include <sys/types.h>
+#include <sys/wait.h>
+
 #if HAVE_FCNTL_H
 #include <fcntl.h>
 #else
